lib/model/om/BaseActor.php

```diff
@@ -641,12 +641,42 @@ public function hasChildren()
 
   public function addAncestorsCriteria(Criteria $criteria)
   {
-    return $criteria->add(QubitActor::LFT, $this->lft, Criteria::LESS_THAN)->add(QubitActor::RGT, $this->rgt, Criteria::GREATER_THAN);
+    if (isset($this->parentId))
+    {
+      $condition = '= '.$this->parentId;
+    }
+    else
+    {
+      $condition = 'IS NULL';
```

Comment:
If `!isset($this->parentId)` then the object does not have any ancestors by definition, in which case I think `return $this` (the self node) is sufficient?

Reply:
Hi @djjuhasz, I struggled a bit whit this too.

We're working with a criteria object on this function, adding a condition and returning that criteria, which is used outside to fetch the resources. At the moment, this function is only used from it's own class but, being public that may change and I didn't want to change its signature.

Returning the criteria directly without applying any condition to it when the resource has no parent will not filter the required resources (all) when the results are fetched. We could use a different condition if the resource has no parent to avoid using the CTE, but the results will be the same.

Moreover, this is quite an edge case as we use a "protected" root resource which should be the parent of all the "top-level" resources on the nested set, so the parent id should only be null for that record.

Reply:
Ah okay, then I think you are right that this is okay as is. :)